libpod/options.go

```diff
@@ -485,6 +485,18 @@ func WithIDMappings(idmappings storage.IDMappingOptions) CtrCreateOption {
 	}
 }
 
+// WithExitCommand sets the ExitCommand for the container, appending on the ctr.ID() to the end
+func WithExitCommand(exitCommand []string) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return ErrCtrFinalized
+		}
+
+		ctr.config.ExitCommand = append(exitCommand, ctr.ID())
```

Comment:
We should make this conditional, some applications might not need it. Add a `appendContainerId bool` argument to `WithExitCommand` maybe?

Reply:
Why would I call WithExitCommand if I did not need it?

Reply:
I can see people wanting to call non-Podman programs after a container exits - maybe sendmail to throw out a "this container has died" message?

Reply:
> I can see people wanting to call non-Podman programs after a container exits...

How about recycling the OCI hook structure for these post-exit commands?  Have `conmon` cann a `podman` subcommand with instructions for finding the post-exit hook JSON, and make `cleanup` one of those entries.  Or wecan address all of that in follow-up work.

Reply:
I think having a dedicated "postexit" API we call would be a good idea (clean up, store the exit code in the DB, and call an OCI extension hook for postexit). I see that as conflicting with this work to a degree - we wouldn't want to let the postexit command be manually specifed through the API, we'd just have them tell us where the executable lives so we can invoke Podman or CRI-O again and start the postexit call.

Reply:
Lets do that later.  For now lets get this merged so we stop leaking mount points all over the place, and we can work on adding advanced features like a poststop hook in later PRs.